lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -4875,6 +4880,11 @@ void ByteCodeGenerator::EmitPropStore(Js::RegSlot rhsLocation, Symbol *sym, Iden
     }
 
     // Arrived at the scope in which the property was defined.
+    if (sym && sym->GetNeedDeclaration() && scope->GetFunc() == funcInfo)
+    {
+        EmitUseBeforeDeclarationRuntimeError(this, Js::Constants::NoRegister);
+    }
```

Comment:
I want to call out here that fall through into the rest of `EmitPropStore` is intentional, whereas if you look at `EmitPropLoad/Typeof` they both stop at this point.

The intention is that this way we don't need `EmitUseBeforeDeclarationRuntimeError` to do something special to make the bytecode well-formed.  Instead we just let `EmitPropStore` do the thing its always done, except it is preceeded by a `RuntimeError` opcode.

We could consider taking this further and removing the `LdUndef` from `EmitUseBeforeDeclarationRuntimeError` and making `EmitPropLoad/Typeof` also continue after emitting the `RuntimeError` bytecode, but I think that is out of scope at this point in the release.  No need to stir that up.